static/js/timerender.js

```diff
@@ -179,13 +179,26 @@ export function render_date(time, time_above, today) {
 }
 
 // Renders the timestamp returned by the <time:> Markdown syntax.
-export function render_markdown_timestamp(time, text) {
+export function render_markdown_timestamp(time) {
     const hourformat = page_params.twenty_four_hour_time ? "HH:mm" : "h:mm a";
     const timestring = format(time, "E, MMM d yyyy, " + hourformat);
-    const titlestring = "This time is in your timezone. Original text was '" + text + "'.";
+    const timezone = new Intl.DateTimeFormat(undefined, {timeZoneName: "short"})
+        .formatToParts(time)
+        .find(({type}) => type === "timeZoneName").value;
+
+    const tz_offset = -time.getTimezoneOffset() / 60;
+    const tz_offset_str =
+        timezone === "UTC" ? "" : " (UTC" + (tz_offset >= 0 ? "+" : "-") + tz_offset + ")";
+
+    const tooltipContent =
+        "Time zone: " +
+        (/GMT[+-][\d:]*/.test(timezone) ? "" : timezone) + // Avoid showing both GMT & UTC
+        tz_offset_str +
+        "<br/>Each user sees this time in their own time zone.";
+
     return {
         text: timestring,
-        title: titlestring,
+        tooltipContent,
```

Comment:
We should use underscore delimiters in variable names, since that's our coding convention.

Reply:
Sorry about that, changed.